packages/react-ui/src/widgets/HistogramWidgetUI/HistogramWidgetUI.js

```diff
@@ -0,0 +1,352 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { useMemo } from 'react';
+import ReactEcharts from '../../custom-components/echarts-for-react';
+import { darken, Grid, Link, makeStyles, Typography, useTheme } from '@material-ui/core';
+import { processFormatterRes } from '../utils/formatterUtils';
+import detectTouchscreen from '../utils/detectTouchScreen';
+import useHistogramInteractivity from './useHistogramInteractivity';
+
+const IS_TOUCH_SCREEN = detectTouchscreen();
+
+const useStyles = makeStyles((theme) => ({
+  optionsSelectedBar: {
+    marginBottom: theme.spacing(2),
+
+    '& .MuiTypography-caption': {
+      color: theme.palette.text.secondary
+    },
+
+    '& .MuiButton-label': {
+      ...theme.typography.caption
+    }
+  },
+  clearButton: {
+    ...theme.typography.caption,
+    cursor: 'pointer'
+  }
+}));
+
+function HistogramWidgetUI({
+  data,
+  ticks,
+  min,
+  max,
+  xAxisFormatter,
+  yAxisFormatter,
+  selectedBars,
+  onSelectedBarsChange,
+  tooltip,
+  tooltipFormatter,
+  animation,
+  filterable: _filterable,
+  height
+}) {
+  const classes = useStyles();
+  const theme = useTheme();
+
+  const filterable = _filterable && !!onSelectedBarsChange;
+
+  const [echartsInstance, setEchartInstance] = useState();
+  const onChartReady = (_echartsInstance) => setEchartInstance(_echartsInstance);
+
+  const formattedData = useMemo(
+    () => formatData(data, ticks, min, max),
+    [data, ticks, min, max]
+  );
+
+  const { onEvents, markAreaOptions } = useHistogramInteractivity({
+    data: formattedData,
+    filterable,
+    selectedBars,
+    onSelectedBarsChange,
+    echartsInstance
+  });
+
+  // Tooltip
+  const tooltipOptions = useMemo(
+    () => ({
+      show: tooltip,
+      trigger: 'item',
+      padding: [theme.spacing(0.5), theme.spacing(1)],
+      borderWidth: 0,
+      textStyle: {
+        ...theme.typography.caption,
+        fontSize: 12,
+        lineHeight: 16,
+        color: theme.palette.common.white
+      },
+      backgroundColor: theme.palette.other.tooltip,
+      confine: true,
+      position: 'top',
+      formatter(params) {
+        return tooltipFormatter(params, xAxisFormatter, yAxisFormatter);
+      }
+    }),
+    [theme, tooltip, tooltipFormatter, xAxisFormatter, yAxisFormatter]
+  );
+
+  // xAxis
+  const xAxisOptions = useMemo(
+    () => ({
+      min,
+      max,
+      interval: (max - min) / formattedData.length,
```

Comment:
We always validate we have >0 here, right?

Reply:
We validate it in HistogramWidget, but not in HistogramWidgetUI

Reply:
I checked that even if formattedData.length is 0 nothing happens.